src/Entities/Minecart.cpp

```diff
@@ -1206,10 +1206,19 @@ void cMinecartWithChest::OpenNewWindow()
 
 void cMinecartWithChest::Destroyed()
 {
-	GetWindow()->OwnerDestroyed();
+	if (GetWindow() != nullptr)
+	{
+		GetWindow()->OwnerDestroyed();
```

Comment:
This code is not good. If the internal storage for the Window is changed by another thread in between these calls, it will still fail.

You should grab a temp copy:
```cpp
auto window = GetWindow();
if (window != nullptr)
{
  window->OwnerDestroyed();
}
```